mbuild/compound.py

```diff
@@ -1662,6 +1662,31 @@ def _visualize_nglview(self, show_ports=False, color_scheme={}):
         overwrite_nglview_default(widget)
         return widget
 
+    def flatten(self):
+        """Flatten the hierarchical structure of the Compound."""
```

Comment:
I added an `inplace` flag for option for this method. There are a few cases that I think would make more sense to have this method in place in stead of returning a clone compound. For example, if you a surface coated with polymer, and each polymer is made up `CH3` and `CH2` children (which contains `C` and `H` particles), this method would allow you to flatten the polymer compound to be a single level, while still remain in contact with the rest of the hierarchy. This would be useful for conversion where you want to consolidate the hierarchical structure (like parmed 3-level `Structure` and `GMSO` future `site.molecule` or `site.residue`.

Reply:
I agree, this looks good to me.